// @contentsquare/react-native-bridge/distribution/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Currency = void 0;
var _reactNative = require("react-native");
var ContentsquareModule = _reactNative.NativeModules.ContentsquareModule;
/**
 * List of the currencies handled.
 * @readonly
 * @enum {number}
 */
var Currency = exports.Currency = {
  AFN: ContentsquareModule.AFN,
  EUR: ContentsquareModule.EUR,
  ALL: ContentsquareModule.ALL,
  DZD: ContentsquareModule.DZD,
  USD: ContentsquareModule.USD,
  AOA: ContentsquareModule.AOA,
  XCD: ContentsquareModule.XCD,
  ARS: ContentsquareModule.ARS,
  AMD: ContentsquareModule.AMD,
  AWG: ContentsquareModule.AWG,
  AUD: ContentsquareModule.AUD,
  AZN: ContentsquareModule.AZN,
  BSD: ContentsquareModule.BSD,
  BHD: ContentsquareModule.BHD,
  BDT: ContentsquareModule.BDT,
  BBD: ContentsquareModule.BBD,
  BYN: ContentsquareModule.BYN,
  BZD: ContentsquareModule.BZD,
  XOF: ContentsquareModule.XOF,
  BMD: ContentsquareModule.BMD,
  INR: ContentsquareModule.INR,
  BTN: ContentsquareModule.BTN,
  BOB: ContentsquareModule.BOB,
  BOV: ContentsquareModule.BOV,
  BAM: ContentsquareModule.BAM,
  BWP: ContentsquareModule.BWP,
  NOK: ContentsquareModule.NOK,
  BRL: ContentsquareModule.BRL,
  BND: ContentsquareModule.BND,
  BGN: ContentsquareModule.BGN,
  BIF: ContentsquareModule.BIF,
  CVE: ContentsquareModule.CVE,
  KHR: ContentsquareModule.KHR,
  XAF: ContentsquareModule.XAF,
  CAD: ContentsquareModule.CAD,
  KYD: ContentsquareModule.KYD,
  CLP: ContentsquareModule.CLP,
  CLF: ContentsquareModule.CLF,
  CNY: ContentsquareModule.CNY,
  COP: ContentsquareModule.COP,
  COU: ContentsquareModule.COU,
  KMF: ContentsquareModule.KMF,
  CDF: ContentsquareModule.CDF,
  NZD: ContentsquareModule.NZD,
  CRC: ContentsquareModule.CRC,
  HRK: ContentsquareModule.HRK,
  CUP: ContentsquareModule.CUP,
  CUC: ContentsquareModule.CUC,
  ANG: ContentsquareModule.ANG,
  CZK: ContentsquareModule.CZK,
  DKK: ContentsquareModule.DKK,
  DJF: ContentsquareModule.DJF,
  DOP: ContentsquareModule.DOP,
  EGP: ContentsquareModule.EGP,
  SVC: ContentsquareModule.SVC,
  ERN: ContentsquareModule.ERN,
  ETB: ContentsquareModule.ETB,
  FKP: ContentsquareModule.FKP,
  FJD: ContentsquareModule.FJD,
  XPF: ContentsquareModule.XPF,
  GMD: ContentsquareModule.GMD,
  GEL: ContentsquareModule.GEL,
  GHS: ContentsquareModule.GHS,
  GIP: ContentsquareModule.GIP,
  GTQ: ContentsquareModule.GTQ,
  GBP: ContentsquareModule.GBP,
  GNF: ContentsquareModule.GNF,
  GYD: ContentsquareModule.GYD,
  HTG: ContentsquareModule.HTG,
  HNL: ContentsquareModule.HNL,
  HKD: ContentsquareModule.HKD,
  HUF: ContentsquareModule.HUF,
  ISK: ContentsquareModule.ISK,
  IDR: ContentsquareModule.IDR,
  XDR: ContentsquareModule.XDR,
  IRR: ContentsquareModule.IRR,
  IQD: ContentsquareModule.IQD,
  ILS: ContentsquareModule.ILS,
  JMD: ContentsquareModule.JMD,
  JPY: ContentsquareModule.JPY,
  JOD: ContentsquareModule.JOD,
  KZT: ContentsquareModule.KZT,
  KES: ContentsquareModule.KES,
  KPW: ContentsquareModule.KPW,
  KRW: ContentsquareModule.KRW,
  KWD: ContentsquareModule.KWD,
  KGS: ContentsquareModule.KGS,
  LAK: ContentsquareModule.LAK,
  LBP: ContentsquareModule.LBP,
  LSL: ContentsquareModule.LSL,
  ZAR: ContentsquareModule.ZAR,
  LRD: ContentsquareModule.LRD,
  LYD: ContentsquareModule.LYD,
  CHF: ContentsquareModule.CHF,
  MOP: ContentsquareModule.MOP,
  MKD: ContentsquareModule.MKD,
  MGA: ContentsquareModule.MGA,
  MWK: ContentsquareModule.MWK,
  MYR: ContentsquareModule.MYR,
  MVR: ContentsquareModule.MVR,
  MRU: ContentsquareModule.MRU,
  MUR: ContentsquareModule.MUR,
  XUA: ContentsquareModule.XUA,
  MXN: ContentsquareModule.MXN,
  MXV: ContentsquareModule.MXV,
  MDL: ContentsquareModule.MDL,
  MNT: ContentsquareModule.MNT,
  MAD: ContentsquareModule.MAD,
  MZN: ContentsquareModule.MZN,
  MMK: ContentsquareModule.MMK,
  NAD: ContentsquareModule.NAD,
  NPR: ContentsquareModule.NPR,
  NIO: ContentsquareModule.NIO,
  NGN: ContentsquareModule.NGN,
  OMR: ContentsquareModule.OMR,
  PKR: ContentsquareModule.PKR,
  PAB: ContentsquareModule.PAB,
  PGK: ContentsquareModule.PGK,
  PYG: ContentsquareModule.PYG,
  PEN: ContentsquareModule.PEN,
  PHP: ContentsquareModule.PHP,
  PLN: ContentsquareModule.PLN,
  QAR: ContentsquareModule.QAR,
  RON: ContentsquareModule.RON,
  RUB: ContentsquareModule.RUB,
  RWF: ContentsquareModule.RWF,
  SHP: ContentsquareModule.SHP,
  WST: ContentsquareModule.WST,
  STN: ContentsquareModule.STN,
  SAR: ContentsquareModule.SAR,
  RSD: ContentsquareModule.RSD,
  SCR: ContentsquareModule.SCR,
  SLL: ContentsquareModule.SLL,
  SGD: ContentsquareModule.SGD,
  XSU: ContentsquareModule.XSU,
  SBD: ContentsquareModule.SBD,
  SOS: ContentsquareModule.SOS,
  SSP: ContentsquareModule.SSP,
  LKR: ContentsquareModule.LKR,
  SDG: ContentsquareModule.SDG,
  SRD: ContentsquareModule.SRD,
  SZL: ContentsquareModule.SZL,
  SEK: ContentsquareModule.SEK,
  CHE: ContentsquareModule.CHE,
  CHW: ContentsquareModule.CHW,
  SYP: ContentsquareModule.SYP,
  TWD: ContentsquareModule.TWD,
  TJS: ContentsquareModule.TJS,
  TZS: ContentsquareModule.TZS,
  THB: ContentsquareModule.THB,
  TOP: ContentsquareModule.TOP,
  TTD: ContentsquareModule.TTD,
  TND: ContentsquareModule.TND,
  TRY: ContentsquareModule.TRY,
  TMT: ContentsquareModule.TMT,
  UGX: ContentsquareModule.UGX,
  UAH: ContentsquareModule.UAH,
  AED: ContentsquareModule.AED,
  USN: ContentsquareModule.USN,
  UYU: ContentsquareModule.UYU,
  UYI: ContentsquareModule.UYI,
  UZS: ContentsquareModule.UZS,
  VUV: ContentsquareModule.VUV,
  VEF: ContentsquareModule.VEF,
  VND: ContentsquareModule.VND,
  YER: ContentsquareModule.YER,
  ZMW: ContentsquareModule.ZMW,
  ZWL: ContentsquareModule.ZWL,
  XBA: ContentsquareModule.XBA,
  XBB: ContentsquareModule.XBB,
  XBC: ContentsquareModule.XBC,
  XBD: ContentsquareModule.XBD,
  XTS: ContentsquareModule.XTS,
  XXX: ContentsquareModule.XXX,
  XAU: ContentsquareModule.XAU,
  XPD: ContentsquareModule.XPD,
  XPT: ContentsquareModule.XPT,
  XAG: ContentsquareModule.XAG
};
