// @contentsquare/react-native-bridge/distribution/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CSMask", {
  enumerable: true,
  get: function get() {
    return _CSMask.CSMask;
  }
});
Object.defineProperty(exports, "CSReliableTarget", {
  enumerable: true,
  get: function get() {
    return _CSReliableTarget.CSReliableTarget;
  }
});
Object.defineProperty(exports, "CSWebView", {
  enumerable: true,
  get: function get() {
    return _CSWebView.CSWebView;
  }
});
Object.defineProperty(exports, "Currency", {
  enumerable: true,
  get: function get() {
    return _constants.Currency;
  }
});
Object.defineProperty(exports, "ErrorAnalysis", {
  enumerable: true,
  get: function get() {
    return _ErrorAnalysis.ErrorAnalysis;
  }
});
exports["default"] = void 0;
var _CSMask = require("./CSMask");
var _CSReliableTarget = require("./CSReliableTarget");
var _CSWebView = require("./CSWebView");
var _ErrorAnalysis = require("./ErrorAnalysis");
var _constants = require("./constants");
var _core = require("./core");
var _sessionReplay = require("./sessionReplay");
var _tracking = require("./tracking");
var _user = require("./user");
/**
 * Contentsquare
 * exports public functions
 */
var _default = exports["default"] = {
  start: _core.start,
  handleUrl: _core.handleUrl,
  getCurrentSessionReplayLink: _sessionReplay.getCurrentSessionReplayLink,
  getUserId: _user.getUserId,
  optIn: _user.optIn,
  optOut: _user.optOut,
  forgetMe: _user.forgetMe,
  resumeTracking: _tracking.resumeTracking,
  stopTracking: _tracking.stopTracking,
  send: _tracking.send,
  sendTransaction: _tracking.sendTransaction,
  sendTransactionWithStringCurrency: _tracking.sendTransactionWithStringCurrency,
  sendDynamicVar: _tracking.sendDynamicVar,
  setDefaultMasking: _sessionReplay.setDefaultMasking,
  sendUserIdentifier: _user.sendUserIdentifier
};
/**
 * CSMask, CSReliableTarget, CSWebView, Currency, ErrorAnalysis
 * @exports CSMask
 * @exports CSWebView
 * @exports CSReliableTarget
 * @exports Currency
 * @exports ErrorAnalysis
 */
